### PR TITLE
Add possibility to restore previous fan settings on exit

### PR DESCRIPTION
**Pull request by Trekky12**
I really appreciate your work for controlling the fans on NAS devices based on the hdd temperatures.
Currently I am testing if the automatic settings on the BIOS or the "manual" settings with your application is more efficient in my use-case.
Unfortunately when hddfancontrol is stopped the fans are set to 100%. This leads to some annoying noises when hddfancontrol is only started for testing purposes. Additionally hddfancontrol is also stopped on shutdown which leads to a full speed fan setting when hddfancontrol is already stopped and the device is still not fully shutdown.

This PR adds the new cli switch `--restore-fan-settings` to restore the fan settings to the values before hddfancontrol was started.

### CODE REVIEW

**Comment by Trekky12** [comment]
I have rebased the PR could you approve the CI? 😊

**Comment by desbma** [comment]
You can ignore the coverage, but the lint errors should be fixed, you can see them here https://github.com/desbma/hddfancontrol/actions/runs/5347735245/jobs/9712459974 or via `pre-commit run -a`.

File: hddfancontrol/__init__.py [hunk 1]
```diff
@@ -759,19 +761,57 @@ def setSpeed(self, target_prct: int) -> None:
...
+
+    def getEnabledValue(self) -> Optional[int]:
+        """Read the enabled value of the fan."""
+        if self.enable_filepath is not None:
+            try:
+                with open(self.enable_filepath, "rt") as enable_file:
+                    enabled_val = int(enable_file.read().strip())
+                    self.logger.debug(f"Fan is in enabled state {enabled_val}")
+                    return enabled_val
+            except FileNotFoundError:
```
Comment (desbma) [inline hddfancontrol/__init__.py:777]: I don't think it is needed to handle this exception.
If `FileNotFoundError` is raised here, it means the file just disappeared while hddfancontrol was running, and this is so unexpected that I'd rather crash than continue.

File: hddfancontrol/__init__.py [hunk 2]
```diff
@@ -759,19 +761,57 @@ def setSpeed(self, target_prct: int) -> None:
...
+                self.enable_filepath = None
+        return None
+
+    def setEnabledValue(self, value: int) -> None:
+        """
+        Set fan enabled value.
+
+                0: no fan speed control (i.e. fan at full speed)
+                1: manual fan speed control enabled
+                2+: automatic fan speed control enabled
```
Comment (desbma) [inline hddfancontrol/__init__.py:787]: Please indent this aligned with the first docstring line.

File: hddfancontrol/__init__.py [hunk 3]
```diff
@@ -1184,7 +1228,7 @@ def cl_main():  # noqa: C901
         type=int,
         nargs=2,
         default=(None, None),
-        help="""CPU temperature range, if CPU temp monitoring is enabled.
+        help="""CPU temperature range (max, critical), if CPU temp monitoring is enabled.
```
Comment (desbma) [inline hddfancontrol/__init__.py:1231]: This change seems unrelated with the rest of the pull request, please remove it.

**Review by desbma — CHANGES_REQUESTED** [review]

File: hddfancontrol/__init__.py (hunk 1 above)
Comment (Trekky12) [reply hddfancontrol/__init__.py:777]: I thought this is needed to address issue #36. This exception is also handled in https://github.com/desbma/hddfancontrol/blob/f57d8b6775e5dd170a014044b78c5fa04f158f49/hddfancontrol/__init__.py#L763-L771

I think `getEnabledValue` is called earlier than `setPwmValue` so this exception should be thrown to not make a breaking change for pwm devices without enable.

What do you think?

File: hddfancontrol/__init__.py (hunk 3 above)
Comment (Trekky12) [reply hddfancontrol/__init__.py:1231]: Yeah that's right, but it took me some time to determine what the two values are.
I will remove it from the PR but can you add this clarification as regular commit?

File: hddfancontrol/__init__.py (hunk 1 above)
Comment (desbma) [reply hddfancontrol/__init__.py:777]: > I think getEnabledValue is called earlier than setPwmValue

I missed that, it's all good then.

File: hddfancontrol/__init__.py (hunk 3 above)
Comment (desbma) [reply hddfancontrol/__init__.py]: Unless you have found a bug (in this case please open an issue), the temperature range set here should be the minimum and maximum temperature the user wants the device to stay in between.

**Comment by desbma** [comment]
Merged, thank you.
